src/passes/Heap2Local.cpp

```diff
@@ -1050,6 +1062,30 @@ struct Array2Struct : PostWalker<Array2Struct> {
                                         builder.makeConst(Literal(result))));
   }
 
+  void visitRefCast(RefCast* curr) {
```

Comment:
Does the comment saying "this is not a problem for ref.cast" above need to be edited?

Reply:
Good catch, fixed.

The cast does go away when we optimize, but now that we allow casts to fail, we do need to know if it fails or not...